python/chemiscope/structures/_stk.py

```diff
@@ -0,0 +1,198 @@
+from collections import Counter
+
+try:
+    import stk
+
+    HAVE_STK = True
+except ImportError:
+    HAVE_STK = False
+
+
+def _stk_valid_structures(
+    frames: stk.Molecule | list[stk.Molecule],
+) -> tuple[list[stk.Molecule], bool]:
+    if HAVE_STK and isinstance(frames, stk.Molecule):
+        # deal with the user passing a single frame
+        return [frames], True
+    elif HAVE_STK and isinstance(frames[0], stk.Molecule):
+        for frame in frames:
+            assert isinstance(frame, stk.Molecule)
+        return frames, True
+    else:
+        return frames, False
+
+
+def _stk_to_json(molecule: stk.Molecule) -> dict[str : int | list]:
+    """Implementation of frame_to_json for stk.Molcule.
+
+    The current implementation assumes no periodic information, which is safe
+    for the majority of stk molecules. If necessary, we can add cell information
+    in the future.
+
+    """
+    pos_mat = molecule.get_position_matrix()
+    data = {}
+    data["size"] = molecule.get_num_atoms()
+    data["names"] = [atom.__class__.__name__ for atom in molecule.get_atoms()]
```

Comment:
this is an interesting way of handling atom names 😃 So if I want to have oxygen and carbon in a structure I need to use `class O: ...` and `class C: ...`?

Reply:
I agree, Lukas figured out a way that works quite nicely, where the class name is resolved based on the required atomic number (https://stk.readthedocs.io/en/stable/_autosummary/stk.Atom.html#stk.Atom). Eg, you can do:

```python
stk.Atom(id=0,atomic_number=6)
# gives object C(0)
stk.C(id=0)
# givies object C(0)
# But,
stk.Atom(id=0,atomic_number=6) == stk.C(id=0)
# False
```

They are all inherited from `stk.Atom` but the name has the element info.

There is no difference in approach, because the interfaces are the same and everything is written to work the immutable class (this goes for all of stk)

Some of this is the result of older code, but it is quite flexible of an approach

Reply:
Not quite as simple for the user as the ase lists, I agree though

Reply:
all good, I think this is a cool way of doing it!